src/hooks/usePairLiquidityMiningCampaigns.ts

```diff
@@ -0,0 +1,165 @@
+import { BigintIsh, Pair } from '@swapr/sdk'
+
+import { gql, useQuery } from '@apollo/client'
+import { useCallback, useMemo } from 'react'
+
+import { SubgraphLiquidityMiningCampaign } from '../apollo'
+import { useAllTokensFromActiveListsOnCurrentChain } from '../state/lists/hooks'
+import {
+  getLowerTimeLimit,
+  getTokenAmount,
+  sortActiveCampaigns,
+  sortExpiredCampaigns,
+  toLiquidityMiningCampaign,
+} from '../utils/liquidityMining'
+import { useKpiTokens } from './useKpiTokens'
+import { useNativeCurrency } from './useNativeCurrency'
+
+import { useActiveWeb3React } from '.'
+
+const CAMPAIGN_REWARDS_TOKEN_COMMON_FIEDLDS = ['address: id', 'name', 'symbol', 'decimals', 'derivedNativeCurrency']
+const CAMPAIGN_COMMON_FIEDLDS = ['duration', 'startsAt', 'endsAt', 'locked', 'stakingCap', 'stakedAmount']
+
+const REGULAR_CAMPAIGN = gql`
+  query($userId: ID) {
+    liquidityMiningCampaigns(first: 999) {
+      address: id
+      ${CAMPAIGN_COMMON_FIEDLDS.join('\r\n')}
+      rewards {
+        token {
+          ${CAMPAIGN_REWARDS_TOKEN_COMMON_FIEDLDS.join('\r\n')}
+        }
+        amount
+      }
+      stakablePair {
+        id
+        reserveNativeCurrency
+        reserveUSD
+        totalSupply
+        reserve0
+        reserve1
+        token0 {
+          address: id
+          name
+          symbol
+          decimals
+        }
+        token1 {
+          address: id
+          name
+          symbol
+          decimals
+        }
+      }
+      liquidityMiningPositions(where: { stakedAmount_gt: 0, user: $userId }) {
+        id
+      }
+    }
+  }
+`
+
+export function usePairLiquidityMiningCampaigns(
+  pair?: Pair
+): {
+  loading: boolean
+  miningCampaigns: any
+} {
+  const pairAddress = useMemo(() => (pair ? pair.liquidityToken.address.toLowerCase() : undefined), [pair])
+  const { chainId, account } = useActiveWeb3React()
+  const subgraphAccountId = useMemo(() => account?.toLowerCase() || '', [account])
```

Comment:
Correct me if I wrong but `useMemo` is not useful here.

Reply:
Yeah, it doesn't look resource intensive. Thanks, I'll remove it.